migrations/inputs_docker/migration.go

```diff
@@ -0,0 +1,188 @@
+	if err := toml.UnmarshalTable(tbl, &plugin); err != nil {
+		return nil, "", err
+	}
+
+	var applied bool
+
+	// 1. Migrate container_names -> container_name_include
+	if containerNamesValue, found := plugin["container_names"]; found {
+		applied = true
+
+		// Convert to []interface{} for easier handling
+		var containerNames []interface{}
+		switch v := containerNamesValue.(type) {
+		case []interface{}:
+			containerNames = v
+		case []string:
+			for _, name := range v {
+				containerNames = append(containerNames, name)
+			}
+		default:
+			return nil, "", fmt.Errorf("container_names value is not a slice: %T", containerNamesValue)
+		}
+
+		// Check if container_name_include already exists
+		if existingInclude, exists := plugin["container_name_include"]; exists {
+			// Merge the arrays
+			var existing []interface{}
+			switch v := existingInclude.(type) {
+			case []interface{}:
+				existing = v
+			case []string:
+				for _, name := range v {
+					existing = append(existing, name)
+				}
+			default:
+				return nil, "", fmt.Errorf("container_name_include value is not a slice: %T", existingInclude)
+			}
+
+			// Append container_names to existing container_name_include
+			merged := append(existing, containerNames...)
+			plugin["container_name_include"] = merged
+		} else {
+			// Create new container_name_include with container_names values
+			plugin["container_name_include"] = containerNames
+		}
+
+		// Remove deprecated field
+		delete(plugin, "container_names")
+	}
+
+	// 2. Migrate perdevice -> perdevice_include
+	if perdeviceValue, found := plugin["perdevice"]; found {
+		// Check if it's a boolean
+		perdeviceBool, ok := perdeviceValue.(bool)
+		if !ok {
+			return nil, "", fmt.Errorf("perdevice value is not a boolean: %T", perdeviceValue)
+		}
+
+		// Only apply migration if perdevice=true, since perdevice=false is default behavior
+		if perdeviceBool {
+			applied = true
+
+			// Check if perdevice_include already exists
+			if existingInclude, exists := plugin["perdevice_include"]; exists {
+				// perdevice=true means include network and blkio if not already present
+				var existing []interface{}
+				switch v := existingInclude.(type) {
+				case []interface{}:
+					existing = v
+				case []string:
+					for _, name := range v {
+						existing = append(existing, name)
+					}
+				default:
+					return nil, "", fmt.Errorf("perdevice_include value is not a slice: %T", existingInclude)
+				}
+
+				// Add network and blkio if not present (following plugin's backward compatibility logic)
+				hasNetwork := false
+				hasBlkio := false
+				for _, item := range existing {
+					if str, ok := item.(string); ok {
+						if str == "network" {
+							hasNetwork = true
+						}
+						if str == "blkio" {
+							hasBlkio = true
+						}
+					}
+				}
+
+				if !hasNetwork {
+					existing = append(existing, "network")
+				}
+				if !hasBlkio {
+					existing = append(existing, "blkio")
+				}
```

Comment:
Wouldn't

```suggestion
				if !slices.Contains(existing, "network") {
					existing = append(existing, "network")
				}

				if !slices.Contains(existing, "blkio") {
					existing = append(existing, "blkio")
				}
```
be easier to read?